chcemvediet/apps/obligees/tests/test_models.py

```diff
@@ -1,369 +1,672 @@
+        oblg = self._create_obligee(name=u'Agency')
+        oblg.name = u'Another Agency'
+        oblg.save(update_fields=[u'name'])
+        self.assertEqual(oblg.slug, u'another-agency')
+        oblg = Obligee.objects.get(pk=oblg.pk)
+        self.assertEqual(oblg.slug, u'another-agency')
+
+    def test_name_genitive_field(self):
+        oblg = self._create_obligee(name_genitive=u'Agency genitive')
+        self.assertEqual(oblg.name_genitive, u'Agency genitive')
+
+    def test_name_genitive_field_default_value_if_omitted(self):
+        oblg = self._create_obligee(omit=[u'name_genitive'])
+        self.assertEqual(oblg.name_genitive, u'')
+
+    def test_name_dative_field(self):
+        oblg = self._create_obligee(name_dative=u'Agency dative')
+        self.assertEqual(oblg.name_dative, u'Agency dative')
+
+    def test_name_dative_field_default_value_if_omitted(self):
+        oblg = self._create_obligee(omit=[u'name_dative'])
+        self.assertEqual(oblg.name_dative, u'')
+
+    def test_name_accusative_field(self):
+        oblg = self._create_obligee(name_accusative=u'Agency accusative')
+        self.assertEqual(oblg.name_accusative, u'Agency accusative')
+
+    def test_name_accusative_field_default_value_if_omitted(self):
+        oblg = self._create_obligee(omit=[u'name_accusative'])
+        self.assertEqual(oblg.name_accusative, u'')
+
+    def test_name_locative_field(self):
+        oblg = self._create_obligee(name_locative=u'Agency locative')
+        self.assertEqual(oblg.name_locative, u'Agency locative')
+
+    def test_name_locative_field_default_value_if_omitted(self):
+        oblg = self._create_obligee(omit=[u'name_locative'])
+        self.assertEqual(oblg.name_locative, u'')
+
+    def test_name_instrumental_field(self):
+        oblg = self._create_obligee(name_instrumental=u'Agency instrumental')
+        self.assertEqual(oblg.name_instrumental, u'Agency instrumental')
+
+    def test_name_instrumental_field_default_value_if_omitted(self):
+        oblg = self._create_obligee(omit=[u'name_instrumental'])
+        self.assertEqual(oblg.name_instrumental, u'')
+
+    def test_gender_field(self):
+        tests = (
+                (u'Test Name 1', Obligee.GENDERS.MASCULINE, _(u'obligees:Obligee:gender:MASCULINE')),
+                (u'Test Name 2', Obligee.GENDERS.FEMININE, _(u'obligees:Obligee:gender:FEMININE')),
+                (u'Test Name 3', Obligee.GENDERS.NEUTER, _(u'obligees:Obligee:gender:NEUTER')),
+                (u'Test Name 4', Obligee.GENDERS.PLURALE, _(u'obligees:Obligee:gender:PLURALE')),
+                )
+        # Make sure we are testing all obligee genders
+        tested_genders = [a for n, a, i in tests]
+        defined_genders = Obligee.GENDERS._inverse.keys()
+        self.assertItemsEqual(tested_genders, defined_genders)
+
+        for name, gender, expected_display in tests:
+            oblg = self._create_obligee(name=name, gender=gender)
```

Comment:
@viliambalaz `name` v `tests` realne nepotrebujes. Kludne mozes pravit nieco ako `name='Test' + str(idx)`, kde index ziskat pomocou `for ... in enumerate(tests)`.

Reply:
:ok: